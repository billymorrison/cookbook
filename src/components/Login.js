import styled from "styled-components";
import foodpictures from "../Media/food.png";

const Login = () => {
  return (
    <MainLoginArea>
      <StyledForm action="submit">
        <h1>Login</h1>
        <StyledLabel>
          Email
          <StyledInput type="email" required />
        </StyledLabel>
        <StyledLabel>
          Password
          <StyledInput type="password" required />
        </StyledLabel>
        <StyledButton type="submit">Login</StyledButton>
        <p>Not a member? Register here.</p>
      </StyledForm>
      <StyledImageSection src={foodpictures} alt="food" />
    </MainLoginArea>
  );
};

export const StyledInput = styled.input`
  border-radius: 5px;
  height: 40px;
  width: 300px;
`;

export const StyledLabel = styled.label`
  padding-bottom: 5px;
  display: flex;
  flex-direction: column;
  height: 75px;
  justify-content: space-between;
  font-size: 20px;
  color: #fff;
`;

export const MainLoginArea = styled.div`
  display: flex;
  width: 60%;
  height: 700px;
  background: yellow;
`;

export const StyledForm = styled.form`
  width: 60%;
  display: flex;
  flex-direction: column;
  background: #f96716;
  color: white;
  align-items: center;
  justify-content: space-evenly;

  h1 {
    color: #fff;
  }
  p {
    color: #fff;
    font-size: 20px;
    font-weight: bold;
  }
`;

export const StyledButton = styled.button`
  width: 305px;
  padding: 10px;
  cursor: pointer;
  font-size: 20px;
  transition: all 0.5s ease;
  &:hover {
    background: #23d997;
    color: #fff;
  }
`;

export const StyledImageSection = styled.img`
  height: 100%;
  width: auto;
  overflow: hidden;
`;

export default Login;
