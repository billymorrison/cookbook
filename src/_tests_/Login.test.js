import ReactDOM from "react-dom";
import Login from "../components/Login";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const setUser = (user) => {
    return user
}

it("renders login component without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Router><Login setUser={setUser}></Login></Router>, div)
});

it("renders login correctly", () => {
  const {getByTestId} = render(<Router><Login setUser={setUser}></Login></Router>);
  expect(getByTestId("login")).toHaveTextContent("Login");
})

it("matches snapshot", () => {
  const tree = renderer.create(<Router><Login setUser={setUser}></Login></Router>).toJSON();
  expect(tree).toMatchSnapshot();
})