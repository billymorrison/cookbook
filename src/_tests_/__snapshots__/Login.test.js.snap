// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-hKgILt cvLgTT"
  data-testid="login"
>
  <form
    action="submit"
    className="sc-eCssSg dlleA-D"
    onSubmit={[Function]}
  >
    <h1>
      Login
    </h1>
    <label
      className="sc-dlfnbm iJivHU"
    >
      Email
      <input
        className="sc-gsTCUz iPtMfY"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </label>
    <label
      className="sc-dlfnbm iJivHU"
    >
      Password
      <input
        className="sc-gsTCUz iPtMfY"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </label>
    <button
      className="sc-gKsewC lmpqCe"
      type="submit"
    >
      Login
    </button>
    <a
      href="/register"
      onClick={[Function]}
    >
      Not a member? 
      <span>
        Register here.
      </span>
    </a>
  </form>
  <img
    alt="food"
    className="sc-iBPRYJ dSUqbp"
    src="default1.png"
  />
</div>
`;
